DEVDOCS-3935 (#1898)
bigcommerce/dev-docs: docs/stencil-docs/developing-further/stored-credit-card-management.mdx

## Update `en.json`
At minimum, a translation for the `en` needs to be added to `lang/en.json: [Pull Request #1603](https://github.com/bigcommerce/cornerstone/pull/1603/files?file-filters%5B%5D=.html&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-b0d4c1fc9d8d2a5a213b27a72cf6c9fe). Add translations for other locales as needed.

## Manage stored account payments

> After adding a stored payment method, shoppers can use the Storefront account payments microapp to manage their stored payment instruments. The Storefront account payments provide a payment method form with billing address fields. Currently, the microapp works only with Ayden.

Reviewer: * Spelling on `Adyen`
* can we link [Storefront account payments microapp]() to somewhere in KB?

Author: I added this callout in [KB-14556](https://bigcommercecloud.atlassian.net/browse/KB-14556)

### Microapp installation instructions

To install the Storefront account microapp to your theme, inject the `account_payments` variable into your template using `add-payment-method.html`.

> ```js showLineNumbers copy

Reviewer: ```suggestion
```handlebars showLineNumbers copy

```

{{#if account_payments}}
    {{{ account_payments }}}
    <script>
        window.BigCommerce = window.BigCommerce || {};
    </script>
{{/if}}
```

> `account_payments` will provide the `<div>` element where you can render Storefront account payments and attach the renderAccountPayments() method to the window.BigCommerce object that displays the Storefront account payments app if the selected payment provider is already implemented.

Reviewer: ```suggestion
The `account_payments` variable includes the `<div>` element that renders the Storefront account payments microapp. You can use a script to attach the `renderAccountPayments()` method to the `window.BigCommerce` object. If the selected payment provider is already implemented, calling `renderAccountPayments()` renders the Storefront account payments microapp.

```

@vitalii-koshovyi I am not sure where the developer attaches `renderAccountPayments()`. Do they add the method on line 99 so that they can call it from a listener?

Reviewer 2: @slsriehl 
The `storefront-account-payments` microapp automatically attaches the `renderAccountPayments()` method to the `window.BigCommerce` object.

The `account_payments` variable is just a `<div>` with the script tag containing the microapp url.
```
<div id="bc-account-payments"></div>

<script type="text/javascript" src="{{ filePath to storefront-account-payments build  }}"></script>
```

So after the script is loaded, storefront-account-payments render method `renderAccountPayments()` is attached to the  `window.BigCommerce`. So theme developer can use it in their script to initialize  the app.

Reviewer 2: @bc-traciporter fyi

> A [Cornerstone example](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/account/add-payment-method.html#L45) shows how to load a script to the page. In this example, we have a condition: if `account_payments` is available - it renders; if not - the fallback payment method form renders.

Reviewer: Make more explicit the logic laid out in these two sentences. There's a lot of information about the relationships between things here that can be clearer.

Author: I added this callout in [KB-14556](https://bigcommercecloud.atlassian.net/browse/KB-14556)

> In the subsequent releases, we'll remove the fallback payment method form because the Storefront account payments microapp will support all payment methods.

Reviewer: rephrase with something like `Our roadmap includes plans for the Storefront accounts payment microapp to support all payment methods, ...( then the fallback payment method form will go away)`

> A [Cornerstone example](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/account/add-payment-method.html#L45) shows how to load a script to the page. In this example, we have a condition because currently, the storefront-account-payments app works only with Adyen: if `account_payments` is available - it renders; if not - the fallback payment method form renders. If you install another payment provider, you still need to render the "add payments" form; therefore, the "add payments method" form displays.

Reviewer: ```suggestion
If the selected payment provider does not have an integration with the microapp, the `account_payments` variable is falsy and the fallback payment form renders.  See an example of the a fallback payment form and the conditional rendering logic from the [Cornerstone theme](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/account/add-payment-method.html#L45).

```

Reviewer 2: `account_payments` variable is empty

The following [Cornerstone example](https://github.com/bigcommerce/cornerstone/blob/master/assets/js/theme/account.js#L102) shows how to initialize the script. In `account.js` we need to call `window.BigCommerce.renderAccountPayments` to initialize the Storefront account payments app with the provided options:

The example has a simple initialize interface.

> ```js showLineNumbers copy

Reviewer: ```suggestion
```ts showLineNumbers copy

```

interface AppConfigInterface {
  styles?: AppStyles;
  storeContextData: StoreContextDataInterface;
  errorHandler(error: string): void;
}
```

* `style` - an optional property for customization of the Storefront account payments form 
* `storeContextData` - a required field where you must provide data from the store context 
* `errorHandler` - a required field where you need to provide the callback that will be called on the Storefront account payments form submission error event.

> ```js showLineNumbers copy

Reviewer: ```suggestion
```ts showLineNumbers copy

```

```
`account_payments` will provide the `<div>` element where you can render Storefront account payments and attach the renderAccountPayments() method to the window.BigCommerce object that displays the Storefront account payments app if the selected payment provider is already implemented.

> A [Cornerstone example](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/account/add-payment-method.html#L45) shows how to load a script to the page. In this example, we have a condition because currently, the storefront-account-payments app works only with Adyen: if `account_payments` is available - it renders; if not - the fallback payment method form renders. If you install another payment provider, you still need to render the "add payments" form; therefore, the "add payments method" form displays.

Reviewer 2: Do we need remove this sentence ```if `account_payments` is available - it renders; if not - the fallback payment method form renders``` ? For me it seems excessiveю
